lib/OnyxCache.js

```diff
@@ -110,7 +111,7 @@ class OnyxCache {
      * @param {Record<string, *>} data - a map of (cache) key - values
      */
     merge(data) {
-        this.storageMap = lodashMerge({}, this.storageMap, data);
+        this.storageMap = lodashMergeWith({}, this.storageMap, data, customizerForMergeWith);
```

Comment:
Since `lodashMergeWith` is always used in conjunction with `customizerForMergeWith`, it seems to me that it would be better to make a merge utility method that hides that a little bit. It would make it easier to work with.